chain-bench with gitlab

<!--
If you have a trouble, feel free to ask.
Make sure you're not asking duplicate question by searching on the issues lists.
-->
Hi! I'm using the SaaS Community Gitlab version, I have the following error.

````
chain-bench -v scan --repository-url https://gitlab.com/krol1/go-cowsay --access-token glpat-xxxxxx
2022-11-22 18:26:07 INF 🚩	Fetch Starting
2022-11-22 18:26:10 INF 🛢️	Fetching Repository Settings Finished
2022-11-22 18:26:10 INF 🌱	Fetching Branch Protection Settings Finished
2022-11-22 18:26:11 WRN failed to fetch approval configuration
2022-11-22 18:26:11 DBG failed to fetch approval configuration error="GET https://gitlab.com/api/v4/projects/41277134/push_rule: 404 {message: 404 Not Found}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x30 pc=0x10151aa04]
````

It seems that the push_rule it's a premium feature: https://docs.gitlab.com/ee/user/project/repository/push_rules.html

--- a/README.md
+++ b/README.md
@@ -169,7 +169,7 @@ It is required to provide an access token with permission to these scopes: `repo
 
 # Supported Providers
 
-We currently support Github as the first SCM, with PAT authentication.
+We currently support Github and Gitlab SCMs, with PAT authentication.
 
 # Please Note
 

--- a/internal/scm-clients/gitlab/mapper.go
+++ b/internal/scm-clients/gitlab/mapper.go
@@ -97,8 +97,8 @@ func toBranch(branch *gitlab.Branch) *models.Branch {
 
 func toBranchProtection(proj *gitlab.Project, protection *gitlab.ProtectedBranch, appConfig *gitlab.ProjectApprovals, appRules []*gitlab.ProjectApprovalRule, pushRules *gitlab.ProjectPushRules) *models.Protection {
 	var p *models.Protection = nil
-	branchRegex := regexp.MustCompile(pushRules.BranchNameRegex)
-	isMatchDefault := branchRegex.Match([]byte(proj.DefaultBranch))
+	branchRegex := regexp.MustCompile(utils.GetValue(pushRules).BranchNameRegex)
+	isMatchDefault := branchRegex.Match([]byte(utils.GetValue(proj).DefaultBranch))
 	approvingReviewCount := 0
 	if len(appRules) > 0 {
 		approvingReviewCount = appRules[0].ApprovalsRequired
@@ -107,13 +107,13 @@ func toBranchProtection(proj *gitlab.Project, protection *gitlab.ProtectedBranch
 		p = &models.Protection{
 			EnforceAdmins: &models.AdminEnforcement{Enabled: false},
 			RequiredPullRequestReviews: &models.PullRequestReviewsEnforcement{
-				DismissStaleReviews:          appConfig.ResetApprovalsOnPush,
-				RequireCodeOwnerReviews:      protection.CodeOwnerApprovalRequired,
+				DismissStaleReviews:          utils.GetValue(appConfig).ResetApprovalsOnPush,
+				RequireCodeOwnerReviews:      utils.GetValue(protection).CodeOwnerApprovalRequired,
 				RequiredApprovingReviewCount: approvingReviewCount},
 			//TODO: Restrictions: toBranchRestrictions(appConfig.Approvers, appConfig.ApproverGroups),
-			AllowForcePushes:               protection.AllowForcePush,
-			RequiredConversationResolution: proj.OnlyAllowMergeIfAllDiscussionsAreResolved,
-			RequiredSignedCommit:           isMatchDefault && pushRules.RejectUnsignedCommits,
+			AllowForcePushes:               utils.GetValue(protection).AllowForcePush,
+			RequiredConversationResolution: utils.GetValue(proj).OnlyAllowMergeIfAllDiscussionsAreResolved,
+			RequiredSignedCommit:           isMatchDefault && utils.GetValue(pushRules).RejectUnsignedCommits,
 			//TODO: PreventSecrets:                 isMatchDefault && pushRules.PreventSecrets,
 		}
 	}

